server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetVacationRequest.java

```diff
@@ -0,0 +1,74 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.model;
+
+import java.util.Map;
+import java.util.Optional;
+
+import org.apache.commons.lang.NotImplementedException;
+import org.apache.james.jmap.api.vacation.Vacation;
+import org.apache.james.jmap.methods.JmapRequest;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import com.google.common.collect.ImmutableMap;
+
+@JsonDeserialize(builder = SetVacationRequest.Builder.class)
+public class SetVacationRequest implements JmapRequest {
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    @JsonPOJOBuilder(withPrefix = "")
+    public static class Builder {
+
+        private ImmutableMap<String, VacationResponse> update;
+
+        public Builder accountId(String accountId) {
+            throw new NotImplementedException();
+        }
+
+        public Builder update(Map<String, VacationResponse> update) {
+            this.update = ImmutableMap.copyOf(update);
+            return this;
+        }
+
+        public SetVacationRequest build() {
+            return new SetVacationRequest(Optional.ofNullable(update).orElse(ImmutableMap.of()));
+        }
+    }
+
+    private final Map<String, VacationResponse> update;
+
+    private SetVacationRequest(Map<String, VacationResponse> update) {
+        this.update = update;
+    }
+
+    public Map<String, VacationResponse> getUpdate() {
+        return update;
+    }
+
+    @JsonIgnore
+    public boolean isValid() {
+        return update.entrySet().size() == 1 && update.containsKey(Vacation.ID);
```

Comment:
Because using a wrong id should generate an error part of the SetVacationResponse, not just an error. To do so, we need to be able to build a SetVacationRequest even if it is using the wrong Id.

See org.apache.james.jmap.methods.integration.SetVacationResponseTest # setVacationResponseShouldContainAnErrorWhenInvalidId